### PR TITLE
Add an API for retrieving a file and translations

### PR DESCRIPTION
**Pull request by MaxSem**
Bug: https://phabricator.wikimedia.org/T206724

### CODE REVIEW

File: src/Service/FileCache.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Service;
+
+/**
+ * This class manages cache for files being translated
+ */
+class FileCache
```
Comment (samwilson) [inline src/Service/FileCache.php]: Could this extend `\Symfony\Component\Cache\Adapter\FilesystemAdapter`? That way we don't have to re-invent things like expiry tracking and we'd be able to e.g. prune expired items via the CLI.

File: src/Service/FileCache.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    private $retriever;
+
+    /** @var string */
+    private $directory;
+
+    /**
+     * @param Retriever $retriever
+     * @param string $directory Cache directory
+     */
+    public function __construct(Retriever $retriever, string $directory = '/tmp')
```
Comment (samwilson) [inline src/Service/FileCache.php]: The directory should default to a subdir of `sys_get_temp_dir()`, but if we use the Symfony cache component that'd be handled automatically.

File: src/Service/Retriever.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Retrieve a file
+     *
+     * @param string $fileName
+     * @return string File contents
+     */
+    public function retrieve(string $fileName): string
+    {
+        $info = $this->api->imageInfo($fileName);
+        $url = reset($info['imageinfo'])['url'];
```
[the remark this answers is not in the document]
Comment (MaxSem) [reply src/Service/Retriever.php]: Done, in MediaWikiApi.

File: src/Controller/ApiController.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     * @return Response
+     */
+    public function getFile(string $fileName): Response
+    {
+        $content = $this->cache->getContent($fileName);
+        return $this->serveContent($content);
+    }
+
+    /**
+     * @Route("/api/translations/{fileName}", name="api_translations")
```
[the remark this answers is not in the document]
Comment (MaxSem) [reply src/Controller/ApiController.php]: Done.

File: src/Service/FileCache.php (hunk 1 above)
Comment (MaxSem) [reply src/Service/FileCache.php]: That's what I started from, however I was frustrated by inability to access the cache file directly, which would make difference when loading huge files.

File: src/Service/FileCache.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+     * @return string
+     */
+    public function getPath(string $fileName): string
+    {
+        $this->tick();
+
+        $path = $this->fullPath($fileName);
+
+        if (!$this->statFile($path)) {
+            $content = $this->retriever->retrieve($fileName);
```
[the remark this answers is not in the document]
Comment (MaxSem) [reply src/Service/FileCache.php]: Done.

File: src/Service/FileCache.php (hunk 2 above)
Comment (MaxSem) [reply src/Service/FileCache.php]: See above.

File: config/services.yaml [hunk 6]
```diff
@@ -29,3 +29,6 @@ services:
 
     # add more service definitions when explicit configuration is needed
     # please note that last definitions always *replace* previous ones
```
[the remark this answers is not in the document]
Comment (MaxSem) [reply config/services.yaml]: Done.

File: src/Service/FileCache.php (hunk 1 above)
Comment (samwilson) [reply src/Service/FileCache.php]: Ah sure, sounds sensible.

File: src/Service/FileCache.php (hunk 2 above)
Comment (samwilson) [reply src/Service/FileCache.php]: Sure. Let's use `sys_get_temp_dir()` then, or maybe better would be a subdirectory in `var/` in the project directory? That's where other cache files are.

Comment (MaxSem) [reply src/Service/FileCache.php]: Made it use `./var` via service config.

Comment (samwilson) [reply src/Service/FileCache.php]: Sorry, I totally overlooked that.

**Pull request merged**